scripts/confix/confix.go

```diff
@@ -0,0 +1,214 @@
+// Program confix applies fixes to a Tendermint TOML configuration file to
+// update a file created with an older version of Tendermint to a compatible
+// format for a newer version.
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/creachadair/atomicfile"
+	"github.com/creachadair/tomledit"
+	"github.com/creachadair/tomledit/parser"
+	"github.com/creachadair/tomledit/transform"
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `Usage: %[1]s [options] -config <path>
+
+Modify the contents of the specified -config TOML file to update the
+names, locations, and values of configuration settings to the current
+configuration layout.
+
+By default, the config file is edited in-place; use -out to write the
+modified file to a different path. In case of any error in updating the
+file, the input is not modified.
+
+Options:
+`, filepath.Base(os.Args[0]))
+		flag.PrintDefaults()
+	}
+}
+
+var (
+	configPath = flag.String("config", "", "Config file path (required)")
+	outPath    = flag.String("out", "", "Output file path (defaults to -input)")
+)
+
+func main() {
+	flag.Parse()
+	if *configPath == "" {
+		log.Fatal("You must specify a non-empty -config path")
+	} else if *outPath == "" {
+		*outPath = *configPath
+	}
```

Comment:
I would strongly prefer this not write to the file by default. I can very clearly envision the case where a user has not checked in their config to vcs and ends up with changes they did not anticipate. We can either write out the change to output, as `gofmt` does _or_ error if it is not specified.

Reply:
That makes sense. I did it this way imitating `go fix`, but you're right that an assumption that holds for code may not hold for config files.

I changed it so that omitting `-out` writes to standard output. The caller can choose to overwrite the input if they want (by setting `-out` == `-input`), but it won't do that by default anymore.